### ISSUE TITLE
a.any() or a.all() truth value error

### ISSUE THREAD

**Issue by andresbh** [coin-or/CyLP#59]
Hi 

based on the example code posted on the [README.rst ](https://github.com/coin-or/CyLP/blob/master/README.rst) I get the following error.
Have you seen it?

I am running it on windows 10 64bit and in an Anaconda Python 2.7 32bit environment. I get the following error prompt. 

Cheers
Andres

```
Traceback (most recent call last):
  File "testCylp.py", line 21, in <module>
    s += A * x <= a
  File "CyClpSimplex.pyx", line 1206, in cylp.cy.CyClpSimplex.CyClpSimplex.__iadd__ (cylp\cy\CyClpSimplex.cpp:15746)
  File "CyClpSimplex.pyx", line 1219, in cylp.cy.CyClpSimplex.CyClpSimplex.addConstraint (cylp\cy\CyClpSimplex.cpp:15965)
  File "d:\Anaconda3\envs\py27_32\lib\site-packages\cylp-0.7.2-py2.7-win32.egg\cylp\py\modeling\CyLPModel.py", line 981, in addConstraint
    c = cons.evaluate(consName)
  File "d:\Anaconda3\envs\py27_32\lib\site-packages\cylp-0.7.2-py2.7-win32.egg\cylp\py\modeling\CyLPModel.py", line 266, in evaluate
    cons.perform(token, left, right)
  File "d:\Anaconda3\envs\py27_32\lib\site-packages\cylp-0.7.2-py2.7-win32.egg\cylp\py\modeling\CyLPModel.py", line 372, in perform
    if left == None:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

The following pip packages are installed

```
(py27_32) D:\>conda list
# packages in environment at d:\Anaconda3\envs\py27_32:
#
certifi                   2017.7.27.1              py27_0    conda-forge
cylp                      0.7.2                     <pip>
mkl                       2017.0.3                      0
numpy                     1.13.1                   py27_0
pip                       9.0.1                    py27_0    conda-forge
python                    2.7.13                        1    conda-forge
scipy                     0.19.1              np113py27_0
setuptools                36.3.0                   py27_0    conda-forge
vs2008_runtime            9.00.30729.5054               0
wheel                     0.29.0                   py27_0    conda-forge
wincertstore              0.2                      py27_0    conda-forge
```

**Comment by andresbh** [comment]
Similarly with a minimalist installation of Python 2.7 32 bit and the relevant pypi packages. I still get that error from above.

I am working with windows 10 and makers build. I am not sure if that windows version might have had a direct effect on cylp. Some months ago it worked well on the computer with windows 10 before that update.

On a Windows 7 machine it works well.

Any ideas?

Thanks in advance

**Comment by andresbh** [comment]
Dear @tkralphs 

have you ever seen this problem?

I haven't been able to solve it yet on the system above.

Cheers 

Andres

**Comment by andresbh** [comment]
it seems cylp parser works only with the version of numpy 1.12 and scipy 0.18

**Comment by Uduse** [comment]
@andresbh the last commit was two years ago... the project is quite quite dead now I guess.

**Comment by tkralphs** [comment]
@andresbh Thanks for reporting this. Is there an easy fix? About the comment of @Uduse, I'm actively working on ways to get CyLP development going again, so it's not dead, as in "abandoned." CyLP does seem to be getting a lot of use and to be of interest to many people and I would like to keep it going. What is needed is a few people who have time to devote to maintaining it. I'm happy to advise anyone who wants pitch in. I just haven't been able to find time myself.

**Comment by andresbh** [comment]
@tkralphs I havent't found any quick solution for that issue. It all came from the changes in numpy anounced with the followin warning

`FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.`

I'd be happy to help out. But am not sure where to start.

**Comment by toliwaga** [comment]
it looks like the fix is simple at the most micro level:

https://github.com/coin-or/CyLP/issues/58

I am not sure how to propagate any updates to a new installable cylp package...

**Comment by tkralphs** [comment]
Thanks @toliwaga. I'll see if I can have a look at it over the holiday weekend.

**Comment by tkralphs** [comment]
This should be fixed by #23 in master.
